core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -928,13 +928,17 @@ private[spark] class TaskSchedulerImpl(
   override def stop(): Unit = {
     speculationScheduler.shutdown()
     if (backend != null) {
-      backend.stop()
+      Utils.tryLogNonFatalError {
+        backend.stop()
```

Comment:
What's the exception you encountered here?

Reply:
when deployed in yarn, org.apache.spark.scheduler.cluster.YarnSchedulerBackend#stop will call requestTotalExecutors() on stop. If the yarn application is killed already, it will receive an IOException on sending the RPC.

Reply:
Checking more, what is the exception thrown in `barrierCoordinator.stop` ?
The should be defensive, and should not have resulted in failures.

Reply:
How about wrapping the others too?